code/particle/util/ParticleProperties.cpp

```diff
@@ -34,8 +34,28 @@ void ParticleProperties::parse(bool nocreate) {
 	}
 }
 
+int ParticleProperties::chooseBitmap()
+{
+	// failsafe check, though we should not have been able to get to this point
+	if (m_bitmap_list.empty())
+		return -1;
+
+	int bitmap_index;
+	int num_bitmaps = (int)m_bitmap_list.size();
+
+	if (num_bitmaps == 1) {
+		bitmap_index = m_bitmap_list[0];
+	} else if (num_bitmaps > 1) {
+		bitmap_index = m_bitmap_list[rand() % num_bitmaps];		
```

Comment:
`rand()` has numerous issues which is why `RandomRange` exists. There you could add a new alias that has a `size_t` as the value type.
That type can then be used in addition to `m_bitmap_list` where you initialize that to return value in the range from 0 to `m_bitmap_list.size - 1`.
Then you could also simplify this code a bit since the random range takes care of most of your cases here.